config/src/main/java/org/springframework/security/config/annotation/web/configurers/oauth2/server/resource/OAuth2ResourceServerConfigurer.java

```diff
@@ -209,13 +231,57 @@ public void configure(H http) throws Exception {
 			return OAuth2ResourceServerConfigurer.this;
 		}
 
+		public JwtConfigurer jwtTimestampValidator(JwtTimestampValidator validator) {
```

Comment:
+1

Reply:
Agreed that I'd also like to find a better approach.

Because we want to add JwtTimestampValidator by default, what does it mean to send a list of validators into addJwtValidators that excludes JwtTimestampValidator?

Finding out that by adding my own custom validator, I inadvertently turned off timestamp validation seems surprising:

```java
.addJwtValidators(new MyIssuerVerifyingValidator()) // now timestamp validation is turned off
```

If this is okay, and either is not surprising, or we are okay with living with that surprise, then my concern is resolved.

If it's not okay then the logic for the opposite case (determining when an element in the list is overriding the default), seems to get tricky.

*Side note* if we have only one method, I still personally prefer `.validator(Validator)`, similar to `.authenticationProvider(AuthenticationProvider)`. I'm not really clear on when one is preferred over the other (XXX over addXXX).